test/1.7_wrap_unwrap_uniV3.js

```diff
@@ -0,0 +1,217 @@
+      token0: await poolContract.token0(),
+      token1: await poolContract.token1(),
+      fee: await poolContract.fee(),
+      tickSpacing: await poolContract.tickSpacing(),
+      maxLiquidityPerTick: await poolContract.maxLiquidityPerTick(),
+    };
+    console.log(PoolImmutables);
+    return PoolImmutables;
+  }
+
+  async function getPoolState(poolContract) {
+    const [liquidity, slot] = await Promise.all([
+      poolContract.liquidity(),
+      poolContract.slot0(),
+    ]);
+
+    const PoolState = {
+      liquidity,
+      sqrtPriceX96: slot[0],
+      tick: slot[1],
+      observationIndex: slot[2],
+      observationCardinality: slot[3],
+      observationCardinalityNext: slot[4],
+      feeProtocol: slot[5],
+      unlocked: slot[6],
+    };
+    console.log(PoolState);
+    return PoolState;
+  }
+
+
+  describe('Test Uni V3 liquidity pool', () => {
+
+    it('Should convert 2 ETH to USDT Token(s) from MakerDao via Uniswap', async () => {
+
+      const zeroAddress = process.env.ZERO_ADDRESS;
+      const userSlippageTolerance = config.userSlippageTolerance;
+      const daiToken = new ethers.Contract(usdtTokenAddress, abi, provider);
+      const poolContract = new ethers.Contract(
+        poolAddress,
+        abi_uniswap,
+        provider
+      );
+      
+      const [immutables, state] = await Promise.all([
+        getPoolImmutables(poolContract),
+        getPoolState(poolContract),
+      ]);
+      const TokenB = new Token(3, immutables.token0, 6, "USDT", "USD Tether");
+
+      const TokenA = new Token(3, immutables.token1, 18, "WETH", "Wrapped Ether");
+
+      const poolExample = new Pool(
+        TokenA,
+        TokenB,
+        immutables.fee,
+        state.sqrtPriceX96.toString(),
+        state.liquidity.toString(),
+        state.tick
+      );
+      console.log(poolExample);
+      const tickLower = nearestUsableTick(state.tick, immutables.tickSpacing) - immutables.tickSpacing * 2;
+      const tickUpper = nearestUsableTick(state.tick, immutables.tickSpacing) + immutables.tickSpacing * 2;
+      console.log("tick lower %s - tick Upper %s", tickLower, tickUpper);
+      const types = ['address', 'uint24', 'address']
+      const values = [
+        wethAddress,
+        3000,
+        usdtTokenAddress]
+      const encodedKey = ethers.utils.defaultAbiCoder.encode(types, values)
+      const incentiveId = ethers.utils.keccak256(encodedKey)
+
+      // Please note, the number of dai tokens we want to get doesn't matter, so the unit amount is just a placeholder
+      const amountPlaceholder = ethers.utils.parseEther(unitAmount)
+
+      // We send 2 ETH to the wrapper for conversion
+      let overrides = {
+        value: ethers.utils.parseEther("3")
+      };
+
+      // Convert the 2 ETH to Dai Token(s)
+      const deadline = Math.floor(new Date().getTime() / 1000) + 1000;
+      const paths = [wethAddress, usdtTokenAddress];
+      const path = [incentiveId, incentiveId];
+      let wrapParams = {
```

Comment:
Split out the conversion and wrapping into 2 different tests too. 1 for converting and the other for wrapping